src/Microsoft.Owin.Security.Authorization/IAuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public interface IAuthorizationDependenciesProvider
+    {
+        Func<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnCreate { get; }
+
+        Action<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnDispose { get; }
```

Comment:
The longer I look at this, the more I think we shouldn't expose dispose functionality.  I can't think of a "best practices" scenario when any of the properties of `AuthorizeDependencies` should be disposable.  Instead, the various interfaces should be implemented in such a way that all disposables are wrapped in `using(IDisposable) {...}` within methods.  If we do decide to allow disposal, `IAuthorizationDependenciesProvider` should inherit from `IDisposable` in order to properly implement the dispose pattern.